pkg/kubelet/prober/results/results_manager.go

```diff
@@ -87,3 +100,21 @@ func (m *manager) Remove(id kubecontainer.ContainerID) {
 	defer m.Unlock()
 	delete(m.cache, id)
 }
+
+func (m *manager) Updates() <-chan Update {
+	if m.updates == nil {
```

Comment:
It's a bit concerning that `m.updates` can be accessed/modified by a different goroutine without lock acquiring.